jswingripples/src/main/java/org/incha/ui/dependency/OpenWithIDEAction.java

```
@@ -21,16 +32,87 @@
+        String classNameParam = node.getFullName();
+        JavaProject project = node.getEig().getJavaProject();
+        final String localfilename = classNameParam.replace(".", File.separator ) + ".java";
+        final List<File> sources = project.getBuildPath().getSources();
+        for (final File file : sources) {
+            try {
+                IoUtils.visitTo(file, new FileVisitor() {
+                    @Override
+                    public void exit(final File t) throws IOException {
+                    }
+                    @Override
+                    public boolean enter(final File t) throws IOException {
+                        if (t.isFile() && t.getPath().indexOf(localfilename)!=-1) {
+                            fileToOpen = t;
+                        }
+                        return true;
+                    }
+                });
+            } catch (IOException ex) {
+                Logger.getLogger(OpenWithIDEAction.class.getName()).log(Level.SEVERE, null, ex);
+            }
+        }
+        System.out.println(fileToOpen.getPath());
+        if (!editFile(fileToOpen)){
+            if (!openFile(fileToOpen)){
+                JOptionPane.showMessageDialog(null, "It is not possible to open the file with the default editor.");
+            }
+        }
+        //Desktop.getDesktop().edit(fileToOpen);
+            //Runtime.getRuntime().exec("open "+fileToOpen.getPath());
+            //Desktop.edit(fileToOpen);
         
     }
+    
+    public boolean editFile(final File file) {
+        if (!Desktop.isDesktopSupported()) {
+          return false;
+        }
+
+        Desktop desktop = Desktop.getDesktop();
+        if (!desktop.isSupported(Desktop.Action.EDIT)) {
+          return false;
+        }
+
+        try {
+          desktop.edit(file);
+        } catch (IOException e) {
+          // Log an error
+          return false;
+        }
+
+        return true;
```

Handle failure/success with Exceptions, instead of returning a boolean.
The same below.